### ISSUE TITLE
Are trigger-in events supported in the bag_to_raw tool?

### ISSUE THREAD

**Issue by AndreasAZiegler** [berndpfrommer/event_array_tools#3]
Are trigger-in events supported in the `bag_to_raw` tool?

**Comment by berndpfrommer** [comment]
The bag-to-raw tool does nothing but just copy the packets into the raw file. Apart from the missing header (which the tool glues on) this should be identical to what you get if you record with e.g. metavision_player.
So yes, the trigger-in events must be in the raw file, there is no filtering of them.
I recommend the "perf" tool, it should show you the rate of triggers and events as you run the driver.

**Comment by AndreasAZiegler** [comment]
Thanks for the clarification.
I actually never managed to get the trigger-in events recorded with `metavision_player` and `metavision_studio` and therefore had to write my own capture application. Given that there is [a callback for the trigger events](https://docs.prophesee.ai/stable/metavision_sdk/modules/driver/api/features.html#_CPPv4N10Metavision10ExtTrigger12add_callbackERK24EventsExtTriggerCallback) I was not sure if the raw callback contains the trigger-in events.

**Comment by berndpfrommer** [comment]
Raw contains everything. I'm surprised to hear you didn't get the trigger events to work with metavision_player. But then I never tried myself. Does it work for you with the ROS driver? What camera do you have, EVK4? I have tested with the SilkEVCam (Gen3 sensor) but have not gotten my hands on a Gen4 camera yet.

**Comment by AndreasAZiegler** [comment]
I now managed to get it working with your ROS driver. The problem I had was that I set `trigger_in_mode` to `enabled` as in the documentation but it should be `external` otherwise the `if` [here](https://github.com/berndpfrommer/metavision_ros_driver/blob/master/src/metavision_wrapper.cpp#L211) doesn't work.
We currently run an EVK1 and EVK2 in a sync setup with external triggers from a FB camera. We recently got two EVK4 cameras but since the sync setup is a bit more involved as I expected it to be, we currently keep working with the EVK1 and EVK2 for the current project.
